adagrams/game.py

```diff
@@ -1,11 +1,99 @@
+import random
+
 def draw_letters():
-    pass
+
+    LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+    }
+
+    the_letters = []
+    for key,value in LETTER_POOL.items():
+        for i in range(value):
+            the_letters.append(key)
+
+    the_hand = random.sample(the_letters, 10)  
+    return the_hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    #copy letter_bank to track what letters are still available to use
+    available_letters = [letter for letter in letter_bank]
+    # check if each letter in the word is available
+    for letter in word:
+        # use .upper() to make sure lowercase input is still tracked
+        if letter.upper() in available_letters:
+            available_letters.remove(letter.upper())
+        else:
+            return False
+    return True
 
 def score_word(word):
-    pass
+    score = 0
+    the_score_dict = { 1: ['A','E','I','O','U','L','N','R','S','T'],
+    2 : ['D','G'],
+    3 : ['B','C','M','P'],
+    4: ['F','H','V','W','Y'],
+    5: ['K'],
+    8: ['J','X'],
+    10: ['Q','Z']
+    }
+    
+    for letter in word:
+        for key, value in the_score_dict.items():
+            if letter.upper() in value:
+                score += key
```

Comment:
This is nominally an O(n) solution, but only because the inner loop is not growing in relation to the input size.  For every letter in `word`, this solution will have to loop through all of the letters in `the_score_dict` in order to find the right key-value pair.  This doesn't take advantage of dictionary lookup as an O(1) operation.  Using a `the_score_dict` dictionary where each letter is a key and the value is the point value would result in a more efficient solution, because then there would be no need for an inner loop (the_score_dict[letter] would return the value for the letter).  In the worst case scenario, if all of the letters of word are 'Z', the inner loop in this solution will have to do 26 comparison operations, but if the_score_dict is set up as {'A':1, 'B':3 ... 'Z':10}, the inner loop is not needed and there's only one operation needed to find the value for the letter.